File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.java
// Last updated: 7/30/2025, 12:22:53 PM
class Solution {
    public int findGCD(int[] nums) {
        int min = Arrays.stream(nums).min().getAsInt();
        int max = Arrays.stream(nums).max().getAsInt();
        while(min!=0){
            int temp = max % min;
            max = min;
            min = temp;
        }
        return max;
    }
}

File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.py
# Last updated: 7/30/2025, 12:22:58 PM
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        high = max(nums)
        low = min(nums)
        while(low!=0):
            high,low = low,high%low
        return high